### ISSUE TITLE
ASF's spellchecker from Fdroid has no uses for Firefox. Firefox removed its spellchecker because Android keyboards had spellcheckers, but all of Android's keyboards removed the spellcheckers because browsers used to have spellcheckers

### ISSUE THREAD

**Issue by ETERNALBLUEbullrun** [AnySoftKeyboard/AnySoftKeyboard#3942]
### Steps to reproduce
1.
Have the bad luck to have wasted your cash on an Android phone
2.
Use browsers
3.
Wish you had Linux or Windows

### Expected behaviour
Browsers (or keyboard apps) should have spellcheckers

### Actual behaviour
Neither have spellcheckers

**Android OS version:**
14
**Device manufacturer and model:**
Google Pixel 6
**List of installed add-ons (like languages, or themes):**

**Comment by ETERNALBLUEbullrun** [comment]
ASF (even ASF's "Terminal" layout) lacks the function keys and arrow keys,
so it was a huge sacrifice to switch to ASF,
was just to use spellcheckers;
no spellcheckers no uses.

**Comment by ferromint** [comment]
1. The title is not the description field.
Keeping it short and precise, reduces the risk of being ignored.
2. Avoid using sardonic descriptions / ranting as they are not entertaining / funny but condescending towards the developers, only distract from your actual request and lower its chance of being addressed even further.
3. There is no need to post a second comment in reply to your initial message just one minute after opening the issue, as there is an edit button in GitHub issues.
4. The app is called AS**K** and not AS**F**.
If you cannot take the time to correctly spell a three-letter acronym or name, why should developers spend any time on this unmotivated / sluggish request?
4. In fact, the given problem description (if it can be even described as such) is not clear at all.
ASK has a spell checker / word prediction that works for most users, based on existing issues.

**Comment by ETERNALBLUEbullrun** [comment]
Apologies for bad grammers.
Am not native to Englishes.
Old browsers had spellcheckers that highlighted/underlined mispelled words.
Firefox now just has this for desktops,
and after ask Firefox developers,
got responses that this was removed from Android because Android keyboards have spellcheckers.
Has anyone got this app's spellcheckers to work with Firefox?
![Screenshot_20240314-115916](https://github.com/AnySoftKeyboard/AnySoftKeyboard/assets/94877072/3031631c-8e0b-4972-a118-95bba79f86a1)
This how Firefox looks if you put mispelled words with AnySoftKeyboard, had options on for spellcheckers

**Comment by ferromint** [comment]
> Apologies for bad grammers
> Am not native to Englishes.

That was not part of my criticism, especially as I am not a native English speaker either.

Your screenshot shows the example of deliberately misspelled text in the URL bar.
Does that mean that your issue is actually refers to misspelled words only for this area of the web browser?
If yes, then my advice (1.) of using a short title for your request is still valid.
Please adjust it accordingly, e.g. "Option to enable spell check for special text fields"

I always disabled Google search suggestions for the address bar.
I am not aware of any browser-built-in spell check for the URL bar even in the past.
Text fields within a website are the only place I know of where browser-built-in spell checkers (e.g. Firefox) can be offered to the user. But also there they are disabled for certain special fields (e.g. password fields) by the developer of a website.
Which browser and which version did have this feature that you are now missing which was not the Google search suggestions feature?

I suspect that also in Android apps spell checking is deliberately disabled for special fields. Not being an Android developer, I further speculate that for each app, each text field gives the keyboard app an indication whether the keyboard's spell check and word suggestion functionality should be enabled for that field. As you can imagine, a spell check or predictive text feature e.g. for URL bars and password fields would do more harm than good.

**Comment by ETERNALBLUEbullrun** [comment]
Firefox for desktops has spellcheckers for all text inputs (not just addresses but all text forms.)
Was told from developers that just Firefox for desktops has spellcheckers because keyboard apps have spellcheckers,
but neither AnySoftKeyboard (nor other keyboard apps) have spellcheckers that work for address bars (searches) or text forms.
What should one do about this?
Firefox autocompletes passwords but that is all, no spellcheckers for address bars (searches) or posts.

**Comment by ferromint** [comment]
> Firefox for desktops has spellcheckers for all text inputs (not just addresses but all text forms.)

This is partially correct at least on my Firefox desktop installations but only for text fields in the browser website canvas / rendering area but **not** for the URL / address bar.

> [...] but neither AnySoftKeyboard (nor other keyboard apps) have spellcheckers that work for address bars [...]

I am not sure if the topic of the URL bar spell check is a language barrier / misunderstanding:
Please show me a documentation or any other example or info about the desktop browser-built-in spell check feature for the URL field which is not the Google search suggestions feature.
I never came across that inside the URL bar.
I am only aware of spell checking inside the actual rendering canvas of the desktop browser, e.g. such as this GitHub issue reply text field I am currently typing in on my PC.

> no spellcheckers for [...] posts

This is the first time you mention a "plausible" text field, e.g. the text message area of a forum or this GitHub issue comment area where ASK spell check feature actually is supposed to work.
Perhaps you have the expectation that ASK checks the entire text after a word has been typed.
ASK's spell check does not work like this.
It only checks and suggested while you are currently typing a word but not afterwards.

**Comment by ETERNALBLUEbullrun** [comment]
Once you disable "Remote search suggestions", Firefox uses "Local search suggestions", so if you have searches for "address" and you put "addrss" this falls under local suggestions.
Just use the address bar examples because was more simple to show,
the biggest of issues is no spellcheckers for forms/posts.

**Comment by ferromint** [comment]
> [...] Once you disable "Remote search suggestions", Firefox uses "Local search suggestions", so if you have searches for "address" and you put "addrss" this falls under local suggestions [...]

The local / offline Firefox desktop URL bar search suggestions feature I know of (from multiple PCs) does **not** have a spell check but simply searches the history, bookmarks open tabs etc. If by accident there is a result with a mistyped word, then it will find which is typically not considered a classical spell check feature.

> [...] Just use the address bar examples because was more simple to show [...]

I see.
The problem is that it is a bad example as the URL bar is a special browser field with no intended spell check apart from above mentioned local website URL suggestions feature which is not a spell checker.

> [...] the biggest of issues is no spellcheckers for forms/posts [...]

You did not refer to my explanation of how the ASK spell check actually works.
Are suggestions displayed in your ASK installation while you are typing a word with at least `x` letters or not?
(I think `x` has a the default minimum suggestions threshold value of 3 letters but I don't remember the default exactly.)

**Comment by menny** [comment]
So, technically, Firefox text field (the URL field) tells AnySoftKeyboard that it _should not_ provide suggestions and corrections. I guess we can disregard that and show the suggestions strip anyway. What do you think?

**Comment by ETERNALBLUEbullrun** [comment]
Yes please could you add suggestions, if not for address bars at least for forms/text boxes

**Comment by ferromint** [comment]
> [...] the biggest of issues is no spellcheckers for forms/posts [...]

> [...] You did not refer to my explanation of how the ASK spell check actually works.
Are suggestions displayed in your ASK installation while you are typing a word with at least x letters or not? [...]

@ETERNALBLUEbullrun
This is now my third attempt to get an answer from you to avoid misunderstandings / wrong expectations ...

**Comment by ferromint** [comment]
> [...] I guess we can disregard that and show the suggestions strip anyway. What do you think? [...]

@menny
There are actually two issues / aspects of this bug report:
- no spell check in URL bar
- misunderstanding of ASK’s spell check capabilities

As long as the issue opener does not answer my repeatedly asked question, it is not clear, whether the main reason for the request are actually wrong expectations about the spell checking capabilities / mechanics of ASK.

It might be that the issue opener's expectation is the wish for having all text in a text field checked and all spelling mistakes underlined at once - e.g. like it is done in Firefox Desktop, Office software, etc.

I think this is why he mentioned the initial situation of "being caught between two stools" or "being given the runaround" between web browser app developers and keyboard app developers.

The most recent statement seems to confirm my suspicion:
> [...] Yes please could you add suggestions, if not for address bars at least for forms/text boxes [...]

If this is not a misunderstanding then why should the issue opener even ask for a feature (suggestions for forms/text boxes) which is already available in ASK?

@ETERNALBLUEbullrun not answering this very basic question makes this issue unnecessarily complicated.

**Comment by ETERNALBLUEbullrun** [comment]
@ferromint
AnySoftKeyboard had no suggestions after tests with bluetooth keyboard and Firefox.

**Comment by ferromint** [comment]
@ETERNALBLUEbullrun
You just answered your own question by providing an important information which you never mentioned before:

You are *not* using ASK in the first place but a Bluetooth keyboard and expect at the same time that ASK checks your spelling ...

I hope you see the contradiction and I guess that this issue can be closed.

The next time, please provide *all* important information to avoid wasting time by chasing a phantom.

I guess this issue can be closed.

If the topic of URL bar suggestions is of interest for anyone then it is best addressed in a dedicated separate issue.

@menny
Actually I do get suggestions in the Firefox Nightly URL bar.

**Comment by ETERNALBLUEbullrun** [comment]
Fix: use Qemu to install desktop Firefox through Ubuntu

**Comment by citizen1277** [comment]
With the exception of the Samsung keyboard, I don't have any apps that provide a proper spell-check anymore on Android 15.  I'm talking about underlining misspelled words, not suggestions in the bar above the keyboard or autocorrect. Autocorrect is an abomination.  The suggestions are helpful, but the  most efficient way to type is to look at your output, not at the keyboard.  Apps like Grammarly are all AI data mines these days, no way am I using anything that complicated and bloated or anything that isn't open source.  This is the only thing about ASK that makes me look at other keyboards.  It really slows me down to have to reread everything closely and check spelling myself.

**Comment by SwuduSusuwu** [comment]
> I don't have any apps that provide a proper spell-check anymore on Android 15.

[*Termux*](https://play.google.com/store/apps/details?id=com.termux&hl=en-US) has [*vim*](https://github.com/termux/termux-packages/tree/master/packages/vim) (installable with `pkg install vim`), which has [this bloom filter which shows red squiggles under mispelled words, as clues](https://vim-dev.vim.narkive.com/dhFp1FoF/spell-checking-using-a-bloom-filter). Also, `vim` [ships with a more simple version of spellcheck](https://vimtricks.com/p/vim-spell-check/).
The problem with `vim`'s spellcheck is that this uses extra steps to move the text from *Termux* to your browser. Also, you cannot click on `vim`'s squiggles for suggestions (such as the desktop versions of most browsers allow you to do).

> Apps like Grammarly are all AI data mines these days, no way am I using anything that complicated and bloated or anything that isn't open source

If you don't want to support those, you should also [disable search suggestions for your browser](https://www.makeuseof.com/tag/how-to-disable-search-suggestions-on-every-major-browser/), plus disable search suggestions for your **OS**. To better target ads (plus enhance "trending searches" algorithms), search suggestions send your unsubmitted text into "*Large Language Model*" [datasets](https://www.tensorflow.org/datasets), which uses up your smartphone's limited upload/download.
- Unless you have a Unix **OS** (such as *Ubuntu*), your **OS**'s search suggestions are closed source
- Most browser search suggestions (except *Mozilla Firefox*) are closed source.
- Most webpages which do their own search suggestions (such as `https://google.com/`) use compressed *JavaScript*, some argue is "open source", but some argue is "closed source").
